ingest/lambdas/bitstreams-downloader.ts

```diff
@@ -0,0 +1,78 @@
+import { z } from 'zod';
+
+import { s3 } from '../../lib/aws-clients';
+import * as dspaceClient from '../../lib/dspace-client';
+import { S3ObjectLocation, safeGetObjectJson } from '../../lib/s3-utils';
+import lambdaClient from '../../lib/lambda-client';
+import { dspaceItemSchema } from '../../lib/dspace-schemas';
+import { getStringFromEnv } from '../../lib/env-utils';
+import { findPDFBitstreamItem } from '../../lib/dspace-item';
+
+const s3EventSchema = z.object({
+  Records: z.array(
+    z.object({
+      s3: z.object({
+        bucket: z.object({
+          name: z.string().min(1),
+        }),
+        object: z.object({
+          key: z.string().min(1),
+        }),
+      }),
+    })
+  ).nonempty(),
+});
+
+export const handler = async (event: unknown) => {
+  const s3Event = s3EventSchema.parse(event);
+
+  const dspaceEndpoint = getStringFromEnv('DSPACE_ENDPOINT');
+  const bitstreamSourceBucket = getStringFromEnv('DOCUMENT_BINARY_BUCKET');
+  const indexerFunction = getStringFromEnv('INDEXER_FUNCTION_NAME');
+
+  try {
+    // Get the DSpace item from S3.
+    const s3Location = new S3ObjectLocation(
+      s3Event.Records[0].s3.bucket.name,
```

Comment:
Are we _certain_ that this event will only ever have one record?

Reply:
Updated to support multiple records. This can probably be parsed out into another non-lambda handler file but it's only used here and wanted to keep the change small for now.